# Node ID
node_id:          0

# Rate to publish control signal
ctrl_rate:        20

# Feedback topic
odom_topic:       "/airsim_node/drone_0/odom_global"


# Points to stop as the drone sweeps the surface

fixed_yaw:        true

# # Interwoven straight line: rememeber to set the drone starting position to 5, 0, 0
# setpoints:
#   [ 0,    0,    2,
#     0,    0,    9,

#    -15.0, 0,    9,
#    -15.0, 2.5,  9,
#    -2.5,  2.5,  9,
#    -2.5,  5.0,  9,
#    -15.0, 5.0,  9,
#    -15.0, 7.5,  9,
#    -2.5,  7.5,  9,
   
#    -2.5,  7.5,  7.5,
#    -2.5,  0.0,  7.5,
#    -5.0,  0.0,  7.5,
#    -5.0,  7.5,  7.5,
#    -7.5,  7.5,  7.5,
#    -7.5,  0.0,  7.5,
#    -10.0, 0.0,  7.5,
#    -10.0, 7.5,  7.5,
#    -12.5, 7.5,  7.5,
#    -12.5, 0.0,  7.5,
#    -15.0, 0.0,  7.5,
#    -15.0, 7.5,  7.5,
    
#    -15.0, 7.5,  6.0,
#    -2.5,  7.5,  6.0,
#    -2.5,  5.0,  6.0,
#    -15.0, 5.0,  6.0,
#    -15.0, 2.5,  6.0,
#    -2.5,  2.5,  6.0,
#    -2.5,  0.0,  6.0,
#    -15.0, 0.0,  6.0,
   
#    -15.0, 0.0,  4.5,
#    -15.0, 7.5,  4.5,
#    -12.5, 7.5,  4.5,
#    -12.5, 0.0,  4.5,
#    -10.0, 0.0,  4.5,
#    -10.0, 7.5,  4.5,
#    -7.5,  7.5,  4.5,
#    -7.5,  0.0,  4.5,
#    -5.0,  0.0,  4.5,
#    -5.0,  7.5,  4.5,
#    -2.5,  7.5,  4.5,
#    -2.5,  0.0,  4.5,
    
#    -2.5,  0.0, -10.0]

# # Interwoven straight line compacted: rememeber to set the drone starting position to 5, 0, 0
# setpoints:
#   [ 0,    0,    2,
#     0,    0,    9,
#    -15.0, 0,    9,
#    -15.0, 2.5,  9,
#    -2.5,  2.5,  9,
#    -2.5,  5.0,  9,
#    -15.0, 5.0,  9,
#    -15.0, 7.5,  9,
#    -2.5,  7.5,  9,
   
#    -2.5,  7.5,  6.0,
#    -2.5,  0.0,  6.0,
#    -5.0,  0.0,  6.0,
#    -5.0,  7.5,  6.0,
#    -7.5,  7.5,  6.0,
#    -7.5,  0.0,  6.0,
#    -10.0, 0.0,  6.0,
#    -10.0, 7.5,  6.0,
#    -12.5, 7.5,  6.0,
#    -12.5, 0.0,  6.0,
#    -15.0, 0.0,  6.0,
#    -15.0, 7.5,  6.0,
   
#    -15.0, 7.5,  3.0,
#    -2.5,  7.5,  3.0,
#    -2.5,  0.0,  3.0,
#    -2.5,  0.0, -10.0]

# # NTU Stylish trajctory: remember to set the drone starting position to -9 -2 0
# setpoints:
#   [ 0,     0,     2,
#     0,     0,     6,
#     0,     4,     6,
#     3,     0,     6,
#     3,     4,     6,
#     7,     4,     6,
#     5,     4,     6,
#     5,    -2,     6,
#     0,    -2,     6,
#    -2,     0,     6,
#    -2,     4,     6,
#     0,     6,     6,
#     7.25,  6,     6,
#     7.25,  0.75,  6,
#     8.25,  0,     6,
#     9.25,  0,     6,
#     10.25, 0.75,  6,
#     10.25, 4,     6,
#     10.25, 4,    -10]

# XZ plane: rememeber to set the drone starting position to 5, 0, 0
setpoints:
  [ 0,  0,  2,
    0,  0,  9,
   -17, 0,  9,
   -17, 0,  7.5,
   -2,  0,  7.5,
   -2,  0,  6,
   -17, 0,  6,
   -17, 0,  4.5,
   -2,  0,  4.5,
   -2,  0,  3,
   -17, 0,  3,
   -17, 0,  1.5,
   -2,  0,  1.5,
   -2,  0, -10 ]

# velocity from stops to stops, in m/s
sweep_velocity:   1.0


# Tolerance to determine if the drone has reached the setpoint
reach_tolerance:  0.05
